cl/opinion_page/views.py

```diff
@@ -827,17 +828,39 @@ def citation_handler(
     if cluster_count == 0 and page.isdigit():
         # Do a second pass for the closest opinion and check if we have
         # a page cite that matches -- if it does give the requested opinion
-        possible_match = (
+
+        possible_match = None
+
+        # Create a list of the closest opinion clusters id and page to the
+        # input citation
+        closest_opinion_clusters = list(
```

Comment:
It took me a bit of time to remember what this code is doing, so I think it just needs some better comments. Github doesn't let me make suggestions on unchanged lines of code, but can we tweak this area of the code to be:

```
    if cluster_count == 0 and page.isdigit():
        # We didn't get an exact match on the volume/reporter/page. Perhaps it's a pincite.
        # Try to find the citation immediately *before* this one in the same book. To do so,
        # Get all the opinions from the book, sort them by page number (in Python, b/c pages
        # can have letters), then find the citation just before the requested one.

        possible_match = None

        # Get all the citations in the book
        closest_opinion_clusters = list(
```

Reply:
can I just ask, what are we going to do with citations with page numbers that have letters.

Reply:
What do you mean? They should sort correctly and more or less work. Can you be more concrete?

Reply:
well. we have an issue inside the lexis and west law citation import - for formats that dont fit.  117A XYZ 13A just to make up something.  

It's probably a problem for down the road but we are probably going to need to update this as soon as we update the citation model.  Presuming we update the citation model.

Reply:
Sorry, I still don't understand. The page is a TextField in the model. What's the problem? What do you mean they don't fit?

Reply:
this won't be working soon `if cluster_count == 0 and page.isdigit():` because we need to move away from page being a digit.

Reply:
OK, now I understand your concern. Yeah, I think that `isdigit()` was put in to fix a bug a while back. @quevon24 could you check in the git blame and see if you can learn why we check for digits here? I suspect the fixes in this PR make the `isdigit` check unnecessary going forward.

Reply:
I think you added isdigit() because the citation handler probably failed with a "wrong" page number (with a letter) and it worked, but now that we have page numbers with letters, we don't need that condition.

That part can be removed now because with natural sort we don't have to worry about whether page numbers contain letters or not.

I made the requested changes